Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +9,234 @@ public static class BalanceStats
+        {
+            var arr = new float[v.Length];
+
+            for (int i = 0; i < v.Length; i++)
+            {
+                arr[i] = float.TryParse(v[i], out float parsed) ? parsed : 0f;
+            }
+            return arr;
+        }
+
+        private static string FormatStrings(string[] currentNames)
+        {
+            if (currentNames.Length == 1)
+            {
+                return currentNames[0];
+            }
+            var formatArray = String.Join(", ", currentNames[..^1]);
+            formatArray += $" and {currentNames[^1]}";
+
+            return formatArray;
+        }
+
+        private static string[] AddStrings(string[] currentNames, string v)
+        {
+            var newArray = new string[currentNames.Length + 1];
+            for (int i = 0; i < currentNames.Length; i++)
+            {
+                newArray[i] = currentNames[i];
+            }
+            newArray[^1] = v;
+
+            return newArray;
         }
 
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+            {
+                return "N/A.";
+            }
+            string[] splitArray;
+            float[] floatArray;
+            int firstIndex = 0;
+
+            do
+            {
+                splitArray = peopleAndBalances[firstIndex].Split(',');
+                floatArray = ConvertStringArrayToFloat(splitArray[1..]);
+                firstIndex++;
+
+            } while (floatArray.Length < 2 && firstIndex < peopleAndBalances.Length);
+
+
+            if (peopleAndBalances.Length == firstIndex && floatArray.Length < 2)
+            {
+                return "N/A.";
+            }
+
+            var names = new[] { splitArray[0] };
+            var biggestLoss = FindHighest(floatArray) - FindLowest(floatArray);
+
+            for (int i = firstIndex; i < peopleAndBalances.Length; i++)
+            {
+                var currentSplitArray = peopleAndBalances[0].Split(',');
+                var currentFloatArray = ConvertStringArrayToFloat(splitArray[1..]);
+
+
+                var currentBiggestLoss = FindHighest(currentFloatArray) - FindLowest(currentFloatArray);
+
+                if (biggestLoss < currentBiggestLoss)
+                {
+                    biggestLoss = currentBiggestLoss;
+                    names = new[] { currentSplitArray[0] };
+                }
+                else if (biggestLoss == currentBiggestLoss)
+                {
+                    AddStrings(names, currentSplitArray[0]);
+                }
+            }
+
+            return $"{FormatStrings(names)} lost the most money. -¤{biggestLoss}.";
+        }
+
+        private static float FindLowest(float[] floatArray)
```

Comment:
Lowest what? It should be lowest balance

Reply:
Changed, FindHeighest changed to FindFighestBalance too